internal/controller/postgrescluster/pgadmin.go

```diff
@@ -302,16 +303,24 @@ func (r *Reconciler) reconcilePGAdminStatefulSet(
 	// set the image pull secrets, if any exist
 	sts.Spec.Template.Spec.ImagePullSecrets = cluster.Spec.ImagePullSecrets
 
-	err := errors.WithStack(r.setControllerReference(cluster, sts))
-
-	if err == nil {
-		pgadmin.Pod(cluster, configmap, &sts.Spec.Template.Spec, dataVolume)
-	}
-	if err == nil {
-		err = errors.WithStack(r.apply(ctx, sts))
+	if err := errors.WithStack(r.setControllerReference(cluster, sts)); err != nil {
+		return errors.WithStack(err)
 	}
```

Comment:
⛏️ Wrapped twice. Doesn't break anything; L307 will appear in the logs, not L306.

```suggestion
	if err := errors.WithStack(r.setControllerReference(cluster, sts)); err != nil {
		return err
	}
```